### PR TITLE
[MiHome] Add Smart Vibration Sensor & Switch aith Accelerometer & more

### PR DESCRIPTION
**Pull request by dimalo**
- Added new Aqara Smart Motion Vibration Sensor
- Added "lastAction" timestamp channel to Cube
- reworked README
- small code improvements (Don't reassign function parameters and cover possible NPE)

Feel free to [download and test the JAR](https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/binding/org.openhab.binding.mihome/2.4.0-SNAPSHOT/org.openhab.binding.mihome-2.4.0-SNAPSHOT.jar)

### CODE REVIEW

File: addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_cube.xml [hunk 1]
```diff
@@ -11,7 +11,8 @@
 		<description>Multifunctional controller equipped with an accelerometer and a gyroscope. Triggers the following
 			actions: move, rotate, rotate, flip 90, flip 180, tap twice, shake air, free fall, alert.</description>
 		<channels>
-			<channel id="action" typeId="action"></channel>
+			<channel id="action" typeId="cubeAction"></channel>
+            <channel id="lastAction" typeId="lastAction"></channel>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_cube.xml]: Small mix of tabs and spaces here

File: addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiBridgeHandler.java [hunk 2]
```diff
@@ -54,7 +56,8 @@
 public class XiaomiBridgeHandler extends ConfigStatusBridgeHandler implements XiaomiSocketListener {
 
     private static final int DISCOVERY_LOCK_TIME_MILLIS = 10000;
-    private static final int READ_ACK_RETENTION_MILLIS = 60 * 60 * 1000; // 2 hours
+    private static final int READ_ACK_RETENTION_MILLIS = 2 * 60 * 60 * 1000; // 2 hours
+    private static final int BRIDGE_CONNECTION_TIMEOUT_MILLIS = 5 * 60 * 1000; // 5 minutes
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiBridgeHandler.java]: What would you think about using the following concept, I think that it reads better especially with big numbers.

```java
private static final int BRIDGE_CONNECTION_TIMEOUT_MILLIS = TimeUnit.MINUTES.toMillis(5);
```

File: addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_cube.xml (hunk 1 above)
Comment (dimalo) [reply addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_cube.xml]: done!

File: addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiBridgeHandler.java (hunk 2 above)
Comment (dimalo) [reply addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiBridgeHandler.java]: Thanks! definetly better...

**Comment by martinvw** [comment]
Thanks! Let's wait for the build to complete to be sure :-)

File: addons/binding/org.openhab.binding.mihome/ESH-INF/thing/channel.xml [hunk 3]
```diff
@@ -188,6 +188,13 @@
 		<state readOnly="true"></state>
 	</channel-type>
 
+    <channel-type id="lastAction">
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.mihome/ESH-INF/thing/channel.xml]: The xml files do not look okay yet. Can you please have a second look?

Comment (dimalo) [reply addons/binding/org.openhab.binding.mihome/ESH-INF/thing/channel.xml]: fixed

Comment (dimalo) [reply addons/binding/org.openhab.binding.mihome/ESH-INF/thing/channel.xml]: ...and rebased

**Comment by dimalo** [comment]
OK now I hope it's final. I reformatted all ESH xmls

**Comment by martinvw** [comment]
Thanks all!

**Comment by floviolleau** [comment]
Hi all,

Thanks for the huge effort here. I'm pretty new in openHab community and I just want to test your job done in that PR. How can I get the jar and test it on my server?

Thanks for your help.

**Comment by dimalo** [comment]
hi @floviolleau, welcome!
[There](https://community.openhab.org/t/mi-xiaomi-smart-home-bindings/4711?u=dimalo) is a topic in the openHAB Forum. Much to read if you want. As this is now merged you can just install the latest snapshot build of openhab - see [docs](https://www.openhab.org/docs/installation/).
Otherwise you can download just the [JAR](https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/binding/org.openhab.binding.mihome/2.4.0-SNAPSHOT/org.openhab.binding.mihome-2.4.0-SNAPSHOT.jar) and put it in the addons folder...

**Comment by floviolleau** [comment]
Hi @dimalo,

Thanks for your help. I finally successfully installed the jar by using the addons folders and it works well.

**Pull request merged**